vmdb/app/models/zone.rb

```diff
@@ -130,7 +130,11 @@ def host_ids
 
   def hosts
     MiqPreloader.preload(self, :ext_management_systems => :hosts)
-    self.ext_management_systems.collect(&:hosts).flatten
+    ext_management_systems.collect(&:hosts).flatten
+  end
+
+  def self.hosts_without_a_zone
+    Host.where(:ems_id => nil).all
```

Comment:
Why have the .all at the end...it might be better to return the actual relation.